api/database_engine.go

```diff
@@ -52,3 +54,33 @@ func (e *EverestServer) UpdateDatabaseEngine(ctx echo.Context, namespace, name s
 	}
 	return e.proxyKubernetes(ctx, namespace, databaseEngineKind, name)
 }
+
+// UpgradeDatabaseEngineOperator upgrades the database engine operator to the specified version.
+func (e *EverestServer) UpgradeDatabaseEngineOperator(ctx echo.Context, namespace string, name string) error {
+	// Parse request body.
+	req := &DatabaseEngineOperatorUpgradeParams{}
+	if err := e.getBodyFromContext(ctx, req); err != nil {
+		e.l.Error(err)
+		return ctx.JSON(http.StatusBadRequest, Error{
+			Message: pointer.ToString(
+				"Could not get DatabaseEngineOperatorUpgradeParams from the request body"),
+		})
+	}
+	// Get existing database engine.
+	dbEngine, err := e.kubeClient.GetDatabaseEngine(ctx.Request().Context(), namespace, name)
+	if err != nil {
+		return err
+	}
+	// Update annotation to start upgrade.
+	annotations := dbEngine.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	annotations[everestv1alpha1.DatabaseOperatorUpgradeAnnotation] = req.TargetVersion
+	dbEngine.SetAnnotations(annotations)
+	_, err = e.kubeClient.UpdateDatabaseEngine(ctx.Request().Context(), namespace, dbEngine)
```

Comment:
Let's have some kind of validation/sanitation for the `targetVersion`, at least check that only the acceptable symbols are used. Bc I bet the method wouldn't pass the next pen testing when one tries to inject some harmful code into the string fields and does not receive a Bad Request response but a 500 😅

Reply:
Done, added semver validation